lib/kadira.js

```diff
@@ -212,23 +212,47 @@ Kadira._send = function (payload, callback) {
 
   function callHTTP() {
     Kadira._postData(endpoint, httpOptions, function(err, response){
-      if(response && response.statusCode === 401) {
-        // do not retry if authentication fails
-        throw new Error('Kadira: AppId, AppSecret combination is invalid');
+      if(!(response && response.statusCode)) {
+        err = err || new Error("Kadira: Unknown Error");
+        return tryAgain(err);
       }
 
-      if(response && response.statusCode == 200) {
+      var status = response.statusCode;
+
+      if(status === 200) {
         if(payloadRetries > 0) {
           logger('connected again and payload sent.')
         }
+
         cleaPayloadRetry();
-        callback && callback();
-      } else {
+        return callback && callback();
+      }
+
+      // do not retry if authentication fails
+      if(status === 401) {
+        throw new Error('Kadira: AppId, AppSecret combination is invalid');
+      }
+
+      if(status < 400 || status > 600) {
+        err = err || new Error("Unknown error: " + status);
+        return tryAgain(err);
+      }
+
+      if(status >= 500 && status < 600) {
         if(!err) {
-          err = new Error("Status code: " + response.statusCode);
+          err = new Error("Kadira: Unknown server error: " + status);
         }
-        tryAgain(err);
+
+        return tryAgain(err);
+      }
+
+      // do not retry for kadira agent errors
+      if(status >= 400 && status < 500) {
+        console.error('Kadira: Unknown kadira agent error:', status);
```

Comment:
Message needs to be just agent error. Not unknown agent error. 

Saying unknown is not good I think.